test/sharness/t0180-pubsub-gossipsub.sh

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+
+test_description="Test pubsub with gossipsub"
+
+. lib/test-lib.sh
+
+# start iptb + wait for peering
+NUM_NODES=5
+test_expect_success 'init iptb' '
+  iptb init -n $NUM_NODES --bootstrap=none --port=0
+'
+
+test_expect_success "enable gossipsub" '
+  for x in $(seq 0 4); do
+    ipfsi $x config Pubsub.Router gossipsub
+  done
+'
+
+# this is just a copy of t0180-pubsub; smell.
+startup_cluster $NUM_NODES --enable-pubsub-experiment
```

Comment:
You should be able to wrap this into a function in the pubsub test and call it with different setup, like https://github.com/ipfs/go-ipfs/blob/master/test/sharness/t0170-dht.sh#L10

Reply:
Up to you. I'm usually not a stickler on DRYing up tests, especially integration tests. We should *never* have to change these anyways ("don't break userspace").

Reply:
so if anyone has strong feelings about it, i can merge into a single test -- but i am otherwise inclined to keep it as is.